src/backend/catalog/gp_matview_aux.c

```diff
@@ -187,7 +189,12 @@ InsertMatviewAuxEntry(Oid mvoid, const Query *viewQuery, bool skipdata)
 	values[Anum_gp_matview_aux_mvname - 1] = NameGetDatum(&mvname);
 
 	values[Anum_gp_matview_aux_has_foreign - 1] = BoolGetDatum(has_foreign);
-	
+
+	matviewRel = table_open(mvoid, NoLock);
+	viewsql = nodeToString((Node *) copyObject(get_matview_query(matviewRel)));
+	table_close(matviewRel, NoLock);
+	values[Anum_gp_matview_aux_view_query - 1] = CStringGetTextDatum(viewsql);
```

Comment:
I wonder when drop a column how to affect `pg_rewrite` and `gp_matview_aux` ?

Reply:
Any cases? DIdn't get it.

Reply:
```sql
CREATE MATERIALIZED VIEW emp_summary AS
SELECT id, name, salary FROM employees;
ALTER TABLE employees DROP COLUMN name;
```

Reply:
You can't drop a column of table depended by other objects. Have a try will show that kind of error.